src/helpers/utils.py

```diff
@@ -0,0 +1,148 @@
+    
+    Args:
+        filename (str): Filename containing a date in mm-dd-yyyy format
+        
+    Returns:
+        Optional[str]: Formatted date as yyyy-mm-dd if found and valid, None otherwise
+    """
+    match = re.search(r'(\d{2})[-_/\\|](\d{2})[-_/\\|](\d{4})', filename)
+    if match:
+        month, day, year = match.groups()
+        try:
+            return datetime.strptime(f"{day}-{month}-{year}", "%d-%m-%Y").strftime("%Y-%m-%d")
+        except ValueError:
+            return None
+    return None
+
+def extract_yyyy_mm_dd(filename: str) -> Optional[str]:
+    """
+    Extract a date from a filename in yyyy-mm-dd format and convert to yyyy-mm-dd.
+    
+    Args:
+        filename (str): Filename containing a date in yyyy-mm-dd format
+        
+    Returns:
+        Optional[str]: Formatted date as yyyy-mm-dd if found and valid, None otherwise
+    """
+    match = re.search(r'(\d{4})[-_/\\|](\d{2})[-_/\\|](\d{2})', filename)
+    if match:
+        year, month, day = match.groups()
+        try:
+            return datetime.strptime(f"{day}-{month}-{year}", "%d-%m-%Y").strftime("%Y-%m-%d")
+        except ValueError:
+            return None
+    return None
+
+def extract_yyyymmdd(filename: str) -> Optional[str]:
+    """
+    Extract a date from a filename in yyyymmdd format and convert to yyyy-mm-dd.
+    
+    Args:
+        filename (str): Filename containing a date in yyyymmdd format
+        
+    Returns:
+        Optional[str]: Formatted date as yyyy-mm-dd if found and valid, None otherwise
+    """
+    match = re.search(r'(\d{8})', filename)
+    if match:
+        try:
+            return datetime.strptime(match.group(0), "%Y%m%d").strftime("%Y-%m-%d")
+        except ValueError:
+            return None
+    return None
+
+def extract_report_date(filename: str) -> Optional[str]:
+    """
+    Extract a date from a filename by trying multiple date formats.
+    
+    Attempts to extract dates in the following formats:
+    - mm-dd-yyyy
+    - yyyy-mm-dd
+    - yyyymmdd
+    - dd-mm-yyyy
+    
+    Args:
+        filename (str): Filename that may contain a date
+        
+    Returns:
+        Optional[str]: Formatted date as yyyy-mm-dd if found and valid, None otherwise
+    """
+    for extractor in [extract_mm_dd_yyyy, extract_yyyy_mm_dd, extract_yyyymmdd,extract_dd_mm_yyyy]:
+        date = extractor(filename)
+        if date:
+            return date
+    return None
+
+def read_file_as_string(file_path: str | Path) -> str:
+    """
+    Read the contents of a file and return it as a string.
+    
+    Args:
+        file_path (str | Path): Path to the file to read
+        
+    Returns:
+        str: Contents of the file as a string
+        
+    Raises:
+        FileNotFoundError: If the file does not exist
+    """
+    file_path = Path(file_path)
+    if not file_path.exists():
+        raise FileNotFoundError(f"File not found: {file_path}")
+
+    with file_path.open("r", encoding="utf-8") as f:
```

Comment:
Alternatively, you can just use `open(file, mode)` which will raise `FileNotFoundError` for you if the file doesn't exist.